lib/Parse/ParseDecl.cpp

```diff
@@ -2462,20 +2462,14 @@ void Parser::consumeDecl(ParserPosition BeginParserPosition,
                          ParseDeclOptions Flags,
                          bool IsTopLevel) {
   SourceLoc CurrentLoc = Tok.getLoc();
+
+  SourceLoc EndLoc = PreviousLoc;
   backtrackToPosition(BeginParserPosition);
   SourceLoc BeginLoc = Tok.getLoc();
-  // Consume tokens up to code completion token.
-  while (Tok.isNot(tok::code_complete, tok::eof))
-    consumeToken();
+  SourceLoc PrevLoc = BeginParserPosition.PreviousLoc;
```

Comment:
Since this is only used in one place, could we avoid creating this local variable?  I find it confusing that we have `EndLoc = PreviousLoc != PrevLoc`, but `BeginParserPosition.PreviousLoc` itself is clearer.